src/plugins/discover/public/application/apps/main/utils/update_search_source.ts

```diff
@@ -32,25 +32,24 @@ export function updateSearchSource(
   }
 ) {
   const { uiSettings, data } = services;
-  const usedSort = getSortForSearchSource(
-    sort,
-    indexPattern,
-    uiSettings.get(SORT_DEFAULT_ORDER_SETTING)
-  );
-  const usedSearchSource = persist ? searchSource : searchSource.getParent()!;
+  const parentSearchSource = persist ? searchSource : searchSource.getParent()!;
 
-  usedSearchSource
+  parentSearchSource
     .setField('index', indexPattern)
     .setField('query', data.query.queryString.getQuery() || null)
     .setField('filter', data.query.filterManager.getFilters());
 
   if (!persist) {
+    const usedSort = getSortForSearchSource(
+      sort,
+      indexPattern,
+      uiSettings.get(SORT_DEFAULT_ORDER_SETTING)
+    );
+    const size = uiSettings.get(SAMPLE_SIZE_SETTING);
     searchSource
       .setField('trackTotalHits', true)
-      .setField('size', uiSettings.get(SAMPLE_SIZE_SETTING))
+      .setField('size', size)
```

Comment:
:art: :question: Since we seem to use a different `size` for 2 of the 3 search sources, would it maybe make sense pulling that out of this method, and just set it also explicitly for the document fetch search source in `fetch_documents`?

Reply:
yes, that makes sense, changed!